src/gtimelog/main.py

```diff
@@ -1483,6 +1509,25 @@ def populate_log(self):
         self.add_footer()
         self.scroll_to_end()
 
+    def _get_log_order_key(self):
+        """
+        Returns a function suitable as key parameter for the ordered function
+
+        The parameter to be sorted is deemed a list item as returned by
+        TimeCollection.grouped_entries
+        """
+        if self.log_order == 'alphanum':
+            return lambda x: x[1]
+        elif self.log_order == 'duration':
+            return lambda x: x[2]
+        elif self.log_order == 'tasks':
+            return lambda x: self.tasks.index(x[1])
+        else:  # 'asis'
+            return None  # more or less equivalent to x[0]
+
+    def set_tasks(self, tasks):
+        self.tasks = tasks
```

Comment:
This is probably the place to build that auxiliary sort structure.

Note to self: review how task list updates are being handled, i.e. will this be called every time the content of the task list change, or do we need to register an observer somewhere?

Reply:
Actually no, this method should just go away.

Reply:
Removed.